experimental/packages/opentelemetry-sdk-metrics-base/src/export/MetricReader.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AggregationTemporality } from './AggregationTemporality';
+import { MetricExporter } from './MetricExporter';
+import { MetricProducer } from './MetricProducer';
+
+// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricreader
+
+export abstract class MetricReader {
+  private _shutdown = false;
+  private _metricProducer?: MetricProducer;
+
+  constructor(private _exporter: MetricExporter) {}
```

Comment:
I think the abstract MetricReader should not put the exporter dependencies on its constructor as a user should be able to implement a MetricReader with multiple underlying exporters. However, I'm not going to change MetricReader in this PR, just a note for https://github.com/open-telemetry/opentelemetry-js/issues/2590.

Reply:
This file is just moved from the root src directory to src/export directory with following changes:
1. re-indented with 2 spaces. 
2. added the basic wire API with MetricProducer and MeterProvider.